### PR TITLE
CCD-22886 Add support for new Archive Status field

### PR DESCRIPTION
**Pull request by jamesoleary123**

### CODE REVIEW

File: ComputeClient/Compute.Client.UnitTests/SampleOutputs/ListSnapshots.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<snapshots xmlns="urn:didata.com:api:cloud:types" pageNumber="1" pageCount="1" totalCount="2" pageSize="250">
-	<snapshot id="43559921" replica="true" archived="false">
```
Comment (jamesoleary123) [inline ComputeClient/Compute.Client.UnitTests/SampleOutputs/ListSnapshots.xml:3]: archived attribute is no longer returned by the API

File: ComputeClient/Compute.Contracts/Snapshot/Snapshot.cs [hunk 2]
```diff
@@ -2437,18 +2451,14 @@ public bool replica
...
     public bool archived
     {
         get
         {
-            return this.archivedField;
-        }
-        set
-        {
-            this.archivedField = value;
+            return !string.IsNullOrEmpty(archiveStatus) && (archiveStatus.ToUpper().Equals("ARCHIVED") || archiveStatus.ToUpper().Equals("DOWNLOAD_IN_PROGRESS"));
```
Comment (jamesoleary123) [inline ComputeClient/Compute.Contracts/Snapshot/Snapshot.cs:2461]: Deriving the legacy "archived" value based on the following mapping.

archiveStatus (new field name)        -> archived (legacy field name being removed)
-----------------------------------------------------------------------------------
null                                                   -> false
AVAILABLE_LOCALLY                        -> false
ARCHIVED                                        -> true
DOWNLOAD_IN_PROGRESS             -> true

File: ComputeClient/Compute.Contracts/Snapshot/Snapshot.cs [hunk 3]
```diff
@@ -2437,18 +2451,14 @@ public bool replica
         }
     }
 
+
     /// <remarks/>
-    [System.Xml.Serialization.XmlAttributeAttribute()]
+    [Obsolete("archived is deprecated, please use archiveStatus instead.")]
```
Comment (jamesoleary123) [inline ComputeClient/Compute.Contracts/Snapshot/Snapshot.cs:2456]: Have marked this method as deprecated so to encourage the use of the archiveStatus method longer term as convience methods like this are not ideal for XSD generated classes.

**Review by jimantz — APPROVED** [review]
Looks good 👍

**Pull request merged**